adminpages/reports/sales.php

```diff
@@ -401,79 +412,133 @@ function pmpro_ShowMonthOrYear()
 		google.charts.load('current', {'packages':['corechart']});
 		google.charts.setOnLoadCallback(drawVisualization);
 		function drawVisualization() {
-
-			var data = google.visualization.arrayToDataTable([
-				[
-					'<?php echo esc_html( $date_function );?>', '<?php _e( 'Renewals', 'paid-memberships-pro' );?>', '<?php echo esc_html( sprintf( __( 'New %s', 'paid-memberships-pro' ), ucwords( $type ) ) );?>', '<?php _e( 'Average*', 'paid-memberships-pro' );?>', 
-				],
-				<?php foreach($cols as $date => $value) { 
-					?>
-					['<?php
-						if ( $period == "monthly" ) {
-							echo esc_html(date_i18n("M", mktime(0,0,0,$date,2)));
-						} else {
-							echo esc_html( $date );
-						}
-					?>', <?php echo esc_html( pmpro_round_price( $value[1] ) );?>, <?php echo esc_html( pmpro_round_price( $value[0] - $value[1] ) );?>, <?php echo esc_html( pmpro_round_price( $average ) );?>,  ], 
+			var dataTable = new google.visualization.DataTable();
```

Comment:
The work in the PR by JarrydLong to bring this out of JS for better escaping practices is definitely something we should consider here -- https://github.com/strangerstudios/paid-memberships-pro/pull/1871/files#diff-fd41fe130e45fdd593becf8e34c099fff45ddb06a41c23ad041d3386fff2228dR308

I can help you get these changes ontop of the work he did over there if you'd like or I can help bring over that structure to this PR separate from his changes.

What would you prefer? @kimcoleman

Reply:
Give me a day to consider his PR and what needs to change in mine. There are a lot of changes here that I would be afraid are lost since we approached using the Google charts API slightly differently.